### PR TITLE
Migration of products list

### PR DESCRIPTION
**Pull request by tomas862**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR migrates product list according to our standarts - in this part it only migrates the display related things and not a single action. To access it for now a temporary url is created **admin-dev/index.php/sell/catalog/products/v2/**
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/10631
| How to test?  | I'll write different test cases and what to test since this page is not going to be migrated all within single PR as it might get too huge. There might be more so we should not miss any

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14267)

### CODE REVIEW

**Comment by tomas862** [comment]
Hi @colinegin ! 

I have a few questions regarding product list migration:

1. In current list when bulk enabling, disabling, duplicating or removing the loading pop-up appears - do we need this in the new list as well? Other migrated lists don't have such functionality.
2. Does "Filter by categories" dropdown is required in new product list as well? If so, then it should look the same?
Screenshot (4)

3. Is this huge icon is needed when there are no products?

Screenshot (5)

currently on new list the default visibility is

Screenshot (6)

4. Related with 2nd question - after we click filter categories the position column appears in product list but its totally different of the usual one we have already implemented - you need to click first, then re-order products and submit the result. Should we apply this or keep the one we already have for all migrated lists

it is right now:
download

everywhere else on migrated lists:
cropped

**Comment by MatShir** [comment]
Hi @tomas862,

> 1. In current list when bulk enabling, disabling, duplicating or removing the loading pop-up appears - do we need this in the new list as well? Other migrated lists don't have such functionality.

If we **can't guarantee** a massive operation like for more than 100 products without loading time, we should keep it. 

> 2. Does "Filter by categories" dropdown is required in new product list as well? If so, then it should look the same?



>    Screenshot (4)

The category filter is a requirement for big shops. What do you think of the UX @TristanLDD ?

> 3. Is this huge icon is needed when there are no products?
> 
> Screenshot (5)

 @TristanLDD can we find some nice fresh design? I don't find any empty state for it.

> currently on new list the default visibility is
> 
> Screenshot (6)


> 4. Related with 2nd question - after we click filter categories the position column appears in product list but its totally different of the usual one we have already implemented - you need to click first, then re-order products and submit the result. Should we apply this or keep the one we already have for all migrated lists

+1 for the new drag and drop system, we should have the same behaviours as you said

> 
> it is right now:
> download
> 
> everywhere else on migrated lists:
> cropped

I flag the PR "Waiting for UX" to notice @TristanLDD for the UX recommendation :)

**Comment by colinegin** [comment]
Hello @tomas862 ,

You will find the recommended empty state for the product listing here : [https://invis.io/5FRLCBENAS4#/358670592_Catalog](https://invis.io/5FRLCBENAS4#/358670592_Catalog). I've updated the issue too.
Let us know if there's any other element missing !

**Comment by matks** [comment]
@tomas862 Is this PR still in WIP or can I review ? or maybe an early review is needed ?

**Comment by tomas862** [comment]
hi @matks - it could be wise to review it and merge it after QA approval - its not like 100 percent finished. Some parts are missing - I'll state them in "How to test" section so QA will be aware which parts will be included.

to fully finish it some side PR's ned to be resolved first:

* https://github.com/PrestaShop/PrestaShop/pull/14619
* https://github.com/PrestaShop/PrestaShop/pull/14528
* https://github.com/PrestaShop/PrestaShop/pull/14542 - this one will allow to handle bulk actions in asynch way . At the time of writing this message it still lacks some dialog loading pop-up functionality.

**Comment by LouiseBonnard** [comment]
Hellooo! One small thing that could be nice, I don't know if this PR can handle this part: when you select a filter ('Stationery', for instance), a new column appears called 'Position' and it allows users to reorder their products. 'Reorder' in the sense of organizing them the way you want. Yet, the term is often misunderstood during translation and it is sometimes turned into 'Order again'. 

Capture d’écran 2019-08-26 à 12 25 40

Could it be possible to change this wording and write it 'Rearrange' (which is, I think, more suitable to the notion of position) or 'Reorganize' instead of 'Reorder'? I hope it will allow a better understanding and a more accurate translation later.

cf. @marionf, @sarahdib

**Comment by tomas862** [comment]
> Hellooo! One small thing that could be nice, I don't know if this PR can handle this part: when you select a filter ('Stationery', for instance), a new column appears called 'Position' and it allows users to reorder their products. 'Reorder' in the sense of organizing them the way you want. Yet, the term is often misunderstood during translation and it is sometimes turned into 'Order again'.
> 
> Capture d’écran 2019-08-26 à 12 25 40
> 
> Could it be possible to change this wording and write it 'Rearrange' (which is, I think, more suitable to the notion of position) or 'Reorganize' instead of 'Reorder'? I hope it will allow a better understanding and a more accurate translation later.
> 
> cf. @marionf, @sarahdib

yes , this feature will be needed for sure but at the moment I dont want to expand this PR any more since it is easier to review from the developers perspective. 

so this PR only migrates:

1. Products display
2. List filters
3. Empty state
4. Bulk actions

PR is ready for review! :tada:

File: src/Core/Grid/Definition/Factory/ProductGridDefinitionFactory.php [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+                        ],
+                    ])
+                    ->setAssociatedColumn('name')
+            )
+            ->add(
+                (new Filter('reference', TextType::class))
+                    ->setTypeOptions([
+                        'required' => false,
+                        'attr' => [
+                            'placeholder' => $this->trans('Search ref.', [], 'Admin.Catalog.Help'),
```
Comment (LouiseBonnard) [inline src/Core/Grid/Definition/Factory/ProductGridDefinitionFactory.php]:
```suggestion
                            'placeholder' => $this->trans('Search reference', [], 'Admin.Actions'),
```

File: src/Core/Grid/Definition/Factory/ProductGridDefinitionFactory.php [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+                        ],
+                    ])
+                    ->setAssociatedColumn('reference')
+            )
+            ->add(
+                (new Filter('category', TextType::class))
+                    ->setTypeOptions([
+                        'required' => false,
+                        'attr' => [
+                            'placeholder' => $this->trans('Search category', [], 'Admin.Catalog.Help'),
```
Comment (LouiseBonnard) [inline src/Core/Grid/Definition/Factory/ProductGridDefinitionFactory.php]:
```suggestion
                            'placeholder' => $this->trans('Search category', [], 'Admin.Actions'),
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *#}
+
+<div class="text-center showcase-list-card">
+  <img class="img-responsive mt-3 img-rtl" src="{{ asset('themes/new-theme/img/empty_state/catalog.png') }}">
+
+  <p class="mt-4 showcase-list-card__header">
+    {{ "Let's start building your catalog!"|trans({}, 'Admin.Catalog.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig]: It is not the same wording in the Empty States document [here](https://docs.google.com/document/d/1e_yu6f3Eq1c430k-a2QA9WLZwbwRSPsIeM_jhp09Jdg/edit#heading=h.xsydpo7gxkar), but it suits me fine this way as well - just make sure you use the right quote marks. And also, I'd write _Let's build your catalog!_ instead.

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  </p>
+
+  <p class="mx-auto showcase-list-card__message">
+    {{ 'Building your catalog is the key step; present your brand here and make visitors feel like shopping on your store. Our advice? Pay extra attention to your product photography, neat pictures could increase your conversion rate by 35%!'|trans({}, 'Admin.Catalog.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig]:
```suggestion
    {{ 'Building your catalog is the key step; present your brand here and make visitors feel like shopping on your store. Our advice? Pay extra attention to your product's images, neat pictures could increase your conversion rate by 35%!'|trans({}, 'Admin.Catalog.Feature') }}
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  </p>
+
+  <div class="mt-4">
+    <a href="{{ documentation_link('product') }}" target="_blank" class="btn btn-outline-secondary mr-1">
+      {{ 'Learn how to manage your catalog'|trans({}, 'Admin.Catalog.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig]:
```suggestion
      {{ 'Learn how to manage your catalog'|trans({}, 'Admin.Actions') }}
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    </a>
+
+    <a href="{{ path('admin_product_new') }}" class="btn btn-primary ml-1">
+      {{ 'Add your first product'|trans({}, 'Admin.Catalog.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig]:
```suggestion
      {{ 'Add your first product'|trans({}, 'Admin.Actions') }}
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    </a>
+  </div>
+  <div class="mt-4">
+    <a href="{{ path('admin_import', {'import_type': 'products'}) }}">
+      {% set fileTypes = ['.csv', '.xls', '.xlsx', '.xlst', '.0ds', '.ots'] %}
+
+      {{ 'Import a list of products (filetypes: %fileTypes%).'|trans({'%fileTypes%' : fileTypes|join(', ')}, 'Admin.Catalog.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig]:
```suggestion
      {{ 'Import a list of products (filetypes: %fileTypes%).'|trans({'%fileTypes%' : fileTypes|join(', ')}, 'Admin.Actions') }}
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig (hunk 4 above)
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig]: In the end, it looks more consistent with the terminology of the page, see below:

Capture d’écran 2020-01-30 à 14 41 55

File: admin-dev/themes/new-theme/js/pages/product/index.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
```
Comment (NeOMakinG) [inline admin-dev/themes/new-theme/js/pages/product/index.js:2]: Please update license header :)

File: admin-dev/themes/new-theme/js/pages/product/index.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+import ExportToSqlManagerExtension from '../../components/grid/extension/export-to-sql-manager-extension';
+import ReloadListActionExtension from '../../components/grid/extension/reload-list-extension';
+import SortingExtension from '../../components/grid/extension/sorting-extension';
+import FiltersResetExtension from '../../components/grid/extension/filters-reset-extension';
+import ColumnTogglingExtension from '../../components/grid/extension/column-toggling-extension';
+import SubmitRowActionExtension from '../../components/grid/extension/action/row/submit-row-action-extension';
+import SubmitBulkExtension from '../../components/grid/extension/submit-bulk-action-extension';
+import BulkActionCheckboxExtension from '../../components/grid/extension/bulk-action-checkbox-extension';
+import FiltersSubmitButtonEnablerExtension
+  from '../../components/grid/extension/filters-submit-button-enabler-extension';
```
Comment (NeOMakinG) [inline admin-dev/themes/new-theme/js/pages/product/index.js:36]: Isn't there a path from webpack ? I think so, should be something like 
```suggestion
  from '@components/grid/extension/filters-submit-button-enabler-extension';
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig [hunk 10]
```diff
@@ -0,0 +1,53 @@
+{#**
+ * 2007-2019 PrestaShop and Contributors
```
Comment (NeOMakinG) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig:2]: Update license header please :)

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Actions/Row/preview_product.html.twig [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
```
Comment (NeOMakinG) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Actions/Row/preview_product.html.twig:21]: here too

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
```
Comment (NeOMakinG) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig:21]: here too

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    </a>
+  </div>
+  <div class="mt-4">
+    <a href="{{ path('admin_import', {'import_type': 'products'}) }}">
+      {% set fileTypes = ['.csv', '.xls', '.xlsx', '.xlst', '.0ds', '.ots'] %}
+
+      {{ 'Import a list of products (filetypes: %fileTypes%).'|trans({'%fileTypes%' : fileTypes|join(', ')}, 'Admin.Actions') }}
+    </a>
+  </div>
+</div>
```
Comment (NeOMakinG) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/EmptyState/product.html.twig:53]: Is this fine ? Git doesn't understand the syntax at all 😮

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/index.html.twig [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
```
Comment (NeOMakinG) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/index.html.twig:21]: License header, at the top too

**Review by NeOMakinG — CHANGES_REQUESTED** [review]
Some minor changes for the front-end part

**Comment by PierreRambaud** [comment]
Hi @tomas862 ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**